### PR TITLE
feat: Introduce Organization Membership Requests

### PR DESCRIPTION
**Pull request by panteliselef**
## New pages
https://docs-preview-516.clerkpreview.com/docs/references/javascript/organization-membership-request
https://docs-preview-516.clerkpreview.com/docs/references/javascript/organization/membership-request

### CODE REVIEW

**Comment by royanger** [comment]
> _ No description provided. _

For other reviewers, new pages are:
https://docs-preview-516.clerkpreview.com/docs/references/javascript/organization-membership-request
https://docs-preview-516.clerkpreview.com/docs/references/javascript/organization/membership-request

File: docs/references/javascript/organization/membership-request.mdx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+```typescript
+function getMembershipRequests(params?: GetMembershipRequestParams): Promise<PaginatedResources<OrganizationMembershipRequest>>;
+```
+
+### `GetMembershipRequestParams`
+
+| Name | Type | Description |
+| --- | --- | --- |
+| `initialPage?` | `number` | A number that can be used to skip the first n-1 pages. For example, if `initialPage` is set to 10, it is will skip the first 9 pages and will fetch the 10th page. |
```
Comment (royanger) [inline docs/references/javascript/organization/membership-request.mdx:22]:
```suggestion
| `initialPage?` | `number` | The page to fetch results from. For example, if `initialPage` is set to 10, it  will skip the first 9 pages and will fetch the 10th page. |
```

File: docs/references/javascript/organization/membership-request.mdx [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+| `pageSize?` | `number` | A number that indicates the maximum number of results that should be returned for a specific page. |
+| `status?` | `'pending' \| 'accepted' \| 'revoked'` | The status an request can have. |
```
Comment (royanger) [inline docs/references/javascript/organization/membership-request.mdx:24]:
```suggestion
| `status?` | `'pending' \| 'accepted' \| 'revoked'` | The status a request can have. |
```

File: docs/references/javascript/organization/membership-request.mdx [hunk 3]
```diff
@@ -0,0 +1,33 @@
+---
+title: Organization Membership Request
+description: These are all methods on the Organization class that allow you to manage the membership requests for an organization.
+---
+
+# Organization Membership Request
+
+These are all methods on the [Organization](/docs/references/javascript/organization/organization) class that allow you to manage the membership requests for an organization.
```
Comment (royanger) [inline docs/references/javascript/organization/membership-request.mdx:8]:
```suggestion
These are methods on the [Organization](/docs/references/javascript/organization/organization) class that allow you to manage the membership requests for an organization.
```

File: docs/references/javascript/organization/membership-request.mdx [hunk 4]
```diff
@@ -0,0 +1,33 @@
+---
+title: Organization Membership Request
+description: These are all methods on the Organization class that allow you to manage the membership requests for an organization.
```
Comment (royanger) [inline docs/references/javascript/organization/membership-request.mdx:3]:
```suggestion
description: These are methods on the Organization class that allow you to manage the membership requests for an organization.
```

**Review by royanger — COMMENTED** [review]
Just a few minor nitpicks. Looks good.

File: docs/manifest.json [hunk 5]
```diff
@@ -501,6 +502,10 @@
         "Organization Domain",
         "/references/javascript/organization-domain"
       ],
+      [
+        "Organization Memberhip Request",
```
Comment (alexisintech) [inline docs/manifest.json:506]:
```suggestion
        "Organization Membership Request",
```

File: docs/references/javascript/organization-membership-request.mdx [hunk 6]
```diff
@@ -0,0 +1,53 @@
+---
+title: OrganizationMembershipRequest
+description: The OrganizationMembershipRequest object is the model that describes the request of a user to join an Organization.
+---
+
+# `OrganizationMembershipRequest`
+
+The `OrganizationMembershipRequest` object is the model that describes the request of a user to join an Organization.
```
Comment (alexisintech) [inline docs/references/javascript/organization-membership-request.mdx:8]:
```suggestion
The `OrganizationMembershipRequest` object is the model that describes the request of a user to join an organization.
```

File: docs/references/javascript/organization-membership-request.mdx [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+| `publicUserData` | [`PublicUserData`](/docs/references/javascript/types/public-user-data) | Public information about the user that this request belongs to. |
+| `createdAt` | `Date` | The date and time the membership request was created. |
+| `updatedAt` | `Date` | The date and time the membership request was updated. |
+
+
+## Methods
+
+## `accept()`
+
+Accepts the request of a user to join the Organization the request refers to.
```
Comment (alexisintech) [inline docs/references/javascript/organization-membership-request.mdx:26]:
```suggestion
Accepts the request of a user to join the organization the request refers to.
```

File: docs/references/javascript/organization-membership-request.mdx [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+
+```typescript
+function accept(): Promise<OrganizationMembershipRequest>;
+```
+
+### Returns
+
+| Type | Description |
+| --- | --- |
+| <code>Promise\<[OrganizationMembershipRequest][mem-request-ref]\></code> | This method returns a `Promise` which resolves with an [`OrganizationMembershipRequest`][mem-request-ref] instance. |
```
Comment (alexisintech) [inline docs/references/javascript/organization-membership-request.mdx:36]: Links to the same exact page, so it's not necessary and simply visual noise.

```suggestion
| <code>Promise\<OrganizationMembershipRequest\></code> | This method returns a `Promise` which resolves with an `OrganizationMembershipRequest` instance. |
```

File: docs/references/javascript/organization-membership-request.mdx [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+
+## `reject()`
+
+Rejects the request of a user to join the Organization the request refers to.
```
Comment (alexisintech) [inline docs/references/javascript/organization-membership-request.mdx:41]:
```suggestion
Rejects the request of a user to join the organization the request refers to.
```

File: docs/references/javascript/organization-membership-request.mdx [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+function reject(): Promise<OrganizationMembershipRequest>;
+```
+
+### Returns
+
+| Type | Description |
+| --- | --- |
+| <code>Promise\<[OrganizationMembershipRequest][mem-request-ref]\></code> | This method returns a `Promise` which resolves with an [`OrganizationMembershipRequest`][mem-request-ref] instance. |
+
+[mem-request-ref]: /docs/references/javascript/organization-membership-request
```
Comment (alexisintech) [inline docs/references/javascript/organization-membership-request.mdx:53]: Links to the same exact page, so it's not necessary and simply visual noise.

```suggestion
| <code>Promise\<OrganizationMembershipRequest\></code> | This method returns a `Promise` which resolves with an `OrganizationMembershipRequest` instance. |
```

File: docs/references/javascript/organization/membership-request.mdx [hunk 11]
```diff
@@ -0,0 +1,33 @@
+---
+title: Organization Membership Request
```
Comment (alexisintech) [inline docs/references/javascript/organization/membership-request.mdx:2]: The other pages in the JS references will be updated in an upcoming JavaScript Docs Revamp. But, essentially, we are using sentence-casing for headings ( see https://github.com/clerk/clerk-docs/blob/main/STYLEGUIDE.md#15-use-sentence-case-for-titles )

```suggestion
title: Organization membership requests
```

File: docs/references/javascript/organization/membership-request.mdx [hunk 12]
```diff
@@ -0,0 +1,33 @@
...
+description: These are all methods on the Organization class that allow you to manage the membership requests for an organization.
+---
+
+# Organization Membership Request
```
Comment (alexisintech) [inline docs/references/javascript/organization/membership-request.mdx:6]:
```suggestion
# Organization membership requests
```

File: docs/references/javascript/organization/membership-request.mdx [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+
+These are all methods on the [Organization](/docs/references/javascript/organization/organization) class that allow you to manage the membership requests for an organization.
+
+## `getMembershipRequests()`
+
+Retrieves a list of domains for the currently active organization.
```
Comment (alexisintech) [inline docs/references/javascript/organization/membership-request.mdx:12]:
```suggestion
Retrieves a list of membership requests for the currently active organization.
```

File: docs/references/javascript/organization/membership-request.mdx (hunk 1 above)
Comment (alexisintech) [reply docs/references/javascript/organization/membership-request.mdx:22]: I don't know if I agree. It's not the page to fetch results from, it's the starting page to begin fetching results from.

File: docs/references/javascript/organization/membership-request.mdx [hunk 14]
```diff
@@ -0,0 +1,33 @@
...
+| --- | --- | --- |
+| `initialPage?` | `number` | A number that can be used to skip the first n-1 pages. For example, if `initialPage` is set to 10, it is will skip the first 9 pages and will fetch the 10th page. |
+| `pageSize?` | `number` | A number that indicates the maximum number of results that should be returned for a specific page. |
+| `status?` | `'pending' \| 'accepted' \| 'revoked'` | The status an request can have. |
+
+### Returns
+
+| Type | Description |
+| --- | --- |
+| <code>Promise\<[PaginatedResponse][pag-res-ref]\<[OrganizationMembershipRequest][mem-request-ref]\>></code> | This method returns a `Promise` which resolves with a list of [`OrganizationMembershipRequest`][mem-request-ref] objects. |
```
Comment (alexisintech) [inline docs/references/javascript/organization/membership-request.mdx:30]: We are now deduping reference links ( see the [styleguide](https://github.com/clerk/clerk-docs/blob/main/STYLEGUIDE.md#13-de-dupe-reference-links) )

```suggestion
| Type | Description |
| --- | --- |
| <code>Promise\<[PaginatedResponse][pag-res-ref]\<[OrganizationMembershipRequest][mem-request-ref]\>></code> | This method returns a `Promise` which resolves with a list of `OrganizationMembershipRequest` objects. |
```

**Review by alexisintech — APPROVED** [review]
One minor change, but it looks great!! Thanks so much, great work 😸💖

**Pull request merged**